packages/next/build/webpack/loaders/next-serverless-loader.ts

```diff
@@ -819,6 +821,21 @@ const nextServerlessLoader: loader.Loader = function () {
                 revalidate: renderOpts.revalidate,
               })
               return null
+            } else if (renderOpts.isRedirect && !_nextData) {
+              const redirect = {
+                destination: renderOpts.pageData.pageProps.__N_REDIRECT,
+                statusCode: renderOpts.pageData.pageProps.__N_REDIRECT_STATUS
+              }
+              const statusCode = getRedirectStatus(redirect)
+
+              if (statusCode === PERMANENT_REDIRECT_STATUS) {
+                res.setHeader('Refresh', \`0;url=\${redirect.destination}\`)
```

Comment:
I don't see a test for this refresh header being set.

Reply:
This is tested in an existing test here https://github.com/vercel/next.js/blob/80fd3d5d17d398dc9e7ca81c4a214a5d1f7b7272/test/integration/gssp-redirect/test/index.test.js#L55

Reply:
Added some additional raw serverless tests to ensure this is covered